k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java

```diff
@@ -319,7 +320,7 @@ public String getMessageId() {
         return mMessageId;
     }
 
-    public void generateMessageId() throws MessagingException {
+    public void generateMessageId(UUIDGenerator uuidGenerator) throws MessagingException {
```

Comment:
Yeah. I didn't do this because MimeMessage is extended a couple of times, which makes this a little ugly and touches a lot of files. I made a commit for it, see if you like it that way.